### PR TITLE
feat: Use gg to generate code for pairs

### PR DESCRIPTION
**Pull request by Xuanwo**
This is a demo PR for evaluations.

Related to https://github.com/beyondstorage/go-storage/issues/607.

- Is go-codegen more friendly than go template?
- Is go-codegen easier to learn/use/understanding?

### CODE REVIEW

File: cmd/definitions/generate.go [hunk 1]
```diff
@@ -42,6 +43,63 @@ func generateGlobal(data *Data) {
...
+		fn.WithParameters(codegen.NewParameter("v", v.Type(), false))
+		fn.WithResults(codegen.NewResult("p", "Pair", false))
+		fn.WithBody(`
+return Pair{
+	Key: "{{ .Name }}",
+	Value: v,
+}
+`, v)
+
+		f.AddFunction(fn)
```
Comment (JinnyYi) [inline cmd/definitions/generate.go:86]: It's more readable and easy to debug. But I wondered whether the whole source file will be complicated for more  complex function body and more complex types?

Comment (xxchan) [reply cmd/definitions/generate.go:86]: What do you mean by "easy to debug"?

Comment (Xuanwo) [reply cmd/definitions/generate.go:86]: > more complex function body and more complex types

It's indeed a question, I can't answer for now (:

Comment (JinnyYi) [reply cmd/definitions/generate.go:86]: I have encountered errors during code generation when using templates before and could not visually locate the error in the template. I think maybe this approach will be easier in dealing with this type of problem？

**Comment by Xuanwo** [comment]
ping @JinnyYi  to take another look.

**Comment by Xuanwo** [comment]
I will extend it to more complex functions like operations  tomorrow.

**Comment by Xuanwo** [comment]
I re-design the API of codegen that inspired by `jen`, please take a look.

The code can't be compiled so fat, I will implement it if it looks good.

**Comment by JinnyYi** [comment]
> I re-design the API of codegen that inspired by `jen`, please take a look.

I think it‘s easy to read and translate into the final generated code.

**Comment by Xuanwo** [comment]
ping @JinnyYi for review.

**Comment by JinnyYi** [comment]
LGTM. Only a small doubt, is it possible or necessary to keep only `Description` for `Pair`?

**Comment by Xuanwo** [comment]
> LGTM. Only a small doubt, is it possible or necessary to keep only `Description` for `Pair`?

Yes, we have to. We use `originalDescription` here because we need to make sure the remaining templates works. We will remove `originalDescription` finally!

**Comment by Xuanwo** [comment]
I will bump a version about gg, please merge this PR after update, thanks!

**Review by JinnyYi — APPROVED** [review]
LGTM.

**Pull request merged**